library/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -132,80 +220,249 @@
+    :rtype: ``dict``
+    """
+
+    net_share_repo = FileOnShare(remote=share_name,
+                                 mount_point=share_mnt,
+                                 creds=UserCredentials(share_user, share_pwd),
+                                 isFolder=True)
+    catalog_path = net_share_repo.new_file(catalog_file_name)
+
+    result = idrac.update_mgr.update_from_repo(catalog_path=catalog_path,
+                                               apply_update=apply_update,
+                                               reboot_needed=reboot,
+                                               job_wait=job_wait)
+
+    return result
 
 
 def update_firmware(idrac, module):
-    """Update firmware from a network share and return the job details."""
-    msg = {}
-    msg['changed'] = False
-    msg['update_status'] = {}
+    """
+    Update firmware from a repository (CIFS, NFS, HTTP, HTTPS, FTP)
+
+    :param idrac: iDRAC handle
+    :type idrac: ``iDRAC Connection Object``
+
+    :param module: Ansible module
+    :type module: ``class AnsibleModule``
+
+    :returns: A tuple containing a result dict and error flag
+    :rtype: ``tuple``
+    """
+
+    result = {}
+    result['changed'] = False
+    result['failed'] = False
+    result['update_status'] = {}
+    err = False
 
     try:
-        upd_share = FileOnShare(remote=module.params['share_name'] + "/" + module.params['catalog_file_name'],
-                                mount_point=module.params['share_mnt'],
-                                isFolder=False,
-                                creds=UserCredentials(
-                                    module.params['share_user'],
-                                    module.params['share_pwd'])
-                                )
+        share_name = module.params['share_name']
+        share_user = module.params.get('share_user')
+        share_pwd = module.params.get('share_pwd')
+        share_mnt = module.params.get('share_mnt')
+        catalog_file_name = module.params['catalog_file_name']
+        apply_update = module.params['apply_update']
+        reboot = module.params['reboot']
+        job_wait = module.params['job_wait']
+        ignore_cert_warning = module.params['ignore_cert_warning']
+
+        # check if valid catalog file
+        if not catalog_file_name.lower().endswith('.xml'):
+            module.fail_json(msg="Invalid catalog file: {0}. Must end with \'.xml\' or \'.XML\' extension".format(catalog_file_name))
 
+        # Temporary Fix for 12G and 13G iDRAC - Use WS-Man API for Firmware
+        # update from a network repository
         idrac.use_redfish = True
-        if '12' in idrac.ServerGeneration or '13' in idrac.ServerGeneration:
+        if any(gen in idrac.ServerGeneration for gen in ['12', '13']):
             idrac.use_redfish = False
 
-        apply_update = True
-        msg['update_status'] = idrac.update_mgr.update_from_repo(upd_share,
```

Comment:
you have declared `result['update_status']` above and using `msg['update_status']`. Recommended to use either of these. Also make sure that this is matching with RETURN declaration. Why to change return type to enable support for HTTPS update source.

Reply:
Not sure what you mean. msg['update_status'] was being used in the old code. I renamed the dict to 'result' since msg['msg'] is not very appealing